courses/beginner/10.md

```diff
@@ -48,207 +41,183 @@ control 'nginx-modules' do
 
-control 'nginx-modules' do
-  impact 1.0
-  title 'NGINX modules'
-  desc 'The required NGINX modules should be installed.'
-end
+::: warning
+If new controls are added to a later version of `my_nginx`, they would not be executed unless explicitly required in this scenario.
+:::
 
-control 'nginx-conf' do
-  impact 1.0
-  title 'NGINX configuration'
-  desc 'The NGINX config file should owned by root, be writable only by owner, and not writeable or and readable by others.'
-end
+### Modifying a Control  
 
-require_controls 'my_nginx_overlay' do
-  control 'first-overlay-control'
-  control 'third-overlay-control'
+Let’s say a particular control from an included profile should still run, but the impact isn’t appropriate? When a control is included or required, it can also be modified!
+
+::: code-tabs
+@tab include_controls
+```ruby
+include_controls 'my_nginx' do
+  control 'nginx-modules' do
+    impact 0.5
+  end
 end
 ```
- 
- 
+
+@tab require_controls
 ```ruby
-# my_nginx_overlay
-control 'first-overlay-control'
-control 'second-overlay-control'
-control 'third-overlay-control'
-control 'fourth-overlay-control'
-control 'fifth-overlay-control'
+require_controls 'my_nginx' do
+  control 'nginx-modules' do
+    impact 0.5
+  end
+  control 'nginx-conf-file'
+end
 ```
 
-Whenever `my_nginx` is executed, in addition to its own controls, it will run only the controls specified in the `require_controls` block. In the case, the following controls would be executed:
-
- - nginx-version
- - nginx-modules
- - nginx-conf
- - first-overlay-control
- - third-overlay-control
-
-Controls `second-overlay-control`, `fourth-overlay-control`, and `fifth-overlay-control` would not be run, just as if they were manually skipped. This method of including specific controls ensures only the controls specified are executed; if new controls are added to a later version of `my_nginx_overlay`, they would not be run.
+:::
 
-And, just the way its possible to modify controls when using `include_controls`, controls can be modified as well.
+In the above example, all included or required controls from `my_nginx` profile are executed. However, should control `nginx-modules` fail, it will be raised with an impact of `0.5` instead of the originally-intended impact of `1.0`.
 
+#### Inputs with profile inheritance
 
-```ruby
-# my_nginx profile  
-control 'nginx-version' do
-  impact 1.0
-  title 'NGINX version'
-  desc 'The required version of NGINX should be installed.'
-end
+In order for a control to run properly, the values for inputs can be overriden to align with your environment. Let's say your environment uses `root` not `admin`, so the input for `nginx-shell-access` control can be updated to measure accordingly.
 
-control 'nginx-modules' do
-  impact 1.0
-  title 'NGINX modules'
-  desc 'The required NGINX modules should be installed.'
-end
+One way of updating the input is by using the `inspec.yml` file:
```

Comment:
Can we be very clear to reference that this would only be appropriate for an overlay and that if you are not using the overlay, you should never update the `inspec.yml` file and instead put all inputs in an `inputs` file? Perhaps link to section 5 here for reference. Or, consider linking them as well to the User Class section 7 where it says "What is the difference between tailoring an InSpec profile with inputs vs. overlays?"

Reply:
Now I realize, we're going down a slippery slope by introducing inputs on this page. Inputs have nothing really to do with profile dependency feature of InSpec. This section is about how to do profile dependencies, nothing more? Inputs are discussed in previous sections, so I'm not sure why we need to confuse matters here.

Reply:
Removed the inputs section